diff --git a/arch/arm/boot/dts/stm32mp157a-dk1.dts b/arch/arm/boot/dts/stm32mp157a-dk1.dts
index f415e58..7b50d26 100644
--- a/arch/arm/boot/dts/stm32mp157a-dk1.dts
+++ b/arch/arm/boot/dts/stm32mp157a-dk1.dts
@@ -13,7 +13,7 @@
 #include "stm32mp15xx-dkx.dtsi"
 
 / {
-	model = "STMicroelectronics STM32MP157A-DK1 Discovery Board";
+	model = "STM32MP157A-DK1 Discovery Board -- Regan:DT patched for OpenPLC+CANFD";
 	compatible = "st,stm32mp157a-dk1", "st,stm32mp157";
 
 	aliases {
@@ -26,3 +26,167 @@
 		stdout-path = "serial0:115200n8";
 	};
 };
+
+&m_can1 {
+       pinctrl-names = "default", "sleep";
+       pinctrl-0 = <&m_can1_pins_b>;
+       pinctrl-1 = <&m_can1_sleep_pins_b>;
+       status = "okay";
+};
+
+
+
+#include <dt-bindings/gpio/gpio.h>
+
+&i2c1 {
+    pcf8574: gpio@27 {
+        compatible = "nxp,pcf8574";
+        reg = <0x27>;
+        gpio-controller;
+        #gpio-cells = <2>;
+    };
+};
+
+/ {
+    aux_lcd: hd44780 {
+        compatible = "hit,hd44780";
+        display-height-chars = <2>;
+        display-width-chars = <16>;
+
+        /* Control Lines */
+        rs-gpios     = <&pcf8574 0 GPIO_ACTIVE_HIGH>;
+        rw-gpios     = <&pcf8574 1 GPIO_ACTIVE_HIGH>; /* Required by driver */
+        enable-gpios = <&pcf8574 2 GPIO_ACTIVE_HIGH>;
+
+        /* 4-bit Data Lines */
+        data-gpios   = <&pcf8574 4 GPIO_ACTIVE_HIGH>,
+                       <&pcf8574 5 GPIO_ACTIVE_HIGH>,
+                       <&pcf8574 6 GPIO_ACTIVE_HIGH>,
+                       <&pcf8574 7 GPIO_ACTIVE_HIGH>;
+
+        /* Backlight Control (Pin 3) */
+        backlight-gpios = <&pcf8574 3 GPIO_ACTIVE_HIGH>;
+    };
+};
+/* =========================================================================
+ * 1. PINMUX CONFIGURATIONS (&pinctrl)
+ * ========================================================================= */
+&pinctrl {
+    /* Digital Inputs - Internal Pull-Down Bias */
+    user_inputs_pins: user_inputs-0 {
+        pins {
+            pinmux = <STM32_PINMUX('A', 8, GPIO)>,   /* IN0 - CN2 Pin 7  (PA8)  */
+                     <STM32_PINMUX('B', 10, GPIO)>,  /* IN1 - CN2 Pin 8  (PB10) */
+                     <STM32_PINMUX('B', 12, GPIO)>,  /* IN2 - CN2 Pin 10 (PB12) */
+                     <STM32_PINMUX('G', 8, GPIO)>,   /* IN3 - CN2 Pin 11 (PG8)  */
+                     <STM32_PINMUX('I', 5, GPIO)>,   /* IN4 - CN2 Pin 12 (PI5)  */
+                     <STM32_PINMUX('D', 7, GPIO)>,   /* IN5 - CN2 Pin 13 (PD7)  */
+                     <STM32_PINMUX('G', 15, GPIO)>,  /* IN6 - CN2 Pin 15 (PG15) */
+                     <STM32_PINMUX('F', 1, GPIO)>;   /* IN7 - CN2 Pin 16 (PF1)  */
+            bias-pull-up; /* Holds floating pins at 3V3 until 0V applied */
+        };
+    };
+user_outputs_pins: user_outputs-0 {
+        pins {
+            pinmux = <STM32_PINMUX('F', 0, GPIO)>,   /* OUT1 - CN2 Pin 18 (PF0)  */
+                     <STM32_PINMUX('F', 9, GPIO)>,   /* OUT2 - CN2 Pin 19 (PF9)  */
+                     <STM32_PINMUX('F', 8, GPIO)>,   /* OUT3 - CN2 Pin 21 (PF8)  */
+                     <STM32_PINMUX('F', 4, GPIO)>,   /* OUT4 - CN2 Pin 22 (PF4)  */
+                     <STM32_PINMUX('F', 7, GPIO)>,   /* OUT5 - CN2 Pin 23 (PF7)  */
+                     <STM32_PINMUX('F', 6, GPIO)>,   /* OUT6 - CN2 Pin 24 (PF6)  */
+                     <STM32_PINMUX('F', 3, GPIO)>,   /* OUT7 - CN2 Pin 26 (PF3)  */
+                     <STM32_PINMUX('F', 15, GPIO)>,  /* OUT8 - CN2 Pin 27 (PF15) */
+                     <STM32_PINMUX('A', 13, GPIO)>,  /* OUT9 - CN2 Pin 26 (PA13) RED LED */
+                     <STM32_PINMUX('H', 7, GPIO)>;   /* OUT10 - CN2 Pin 27 (PH7) ORANGE LED */
+            drive-push-pull;
+            bias-pull-down;
+        };
+    };
+
+    /* ADC Pinmux (ADC1_INP2 on PF11) */
+    adc1_pins_a: adc1-pins-0 {
+        pins {
+            pinmux = <STM32_PINMUX('F', 11, ANALOG)>;
+        };
+    };
+
+    adc1_sleep_pins_a: adc1-sleep-pins-0 {
+        pins {
+            pinmux = <STM32_PINMUX('F', 11, ANALOG)>; /* ADC1_INP2 - CN2 Pin 40 */
+            bias-high-z;
+        };
+    };
+};
+
+/* =========================================================================
+ * 2. HARDWARE PERIPHERALS ACTIVATION
+ * ========================================================================= */
+/* Activation (PWM3 PC7 on Pin33 and PWM5 PH11 on pin31 ) */
+&timers3 {
+    status = "okay";
+
+    pwm {
+        pinctrl-names = "default", "sleep";
+        pinctrl-0 = <&pwm3_pins_a>;
+        pinctrl-1 = <&pwm3_sleep_pins_a>;
+        status = "okay";
+    };
+};
+
+&timers5 {
+    status = "okay";
+
+    pwm {
+        pinctrl-names = "default", "sleep";
+        pinctrl-0 = <&pwm5_pins_a>;
+        pinctrl-1 = <&pwm5_sleep_pins_a>;
+        status = "okay";
+    };
+};
+
+/* ADC1 Activation (PF11 on Pin 40) */
+&adc {
+    pinctrl-names = "default", "sleep";
+    pinctrl-0 = <&adc1_pins_a>;
+    pinctrl-1 = <&adc1_sleep_pins_a>;
+    status = "okay";
+
+    adc1: adc@0 {
+        status = "okay";
+        /* Enable channel 2 (which corresponds to PF11 / ADC1_INP2) */
+        st,adc-channels = <2>;
+    };
+};
+
+/* =========================================================================
+ * 3. USER GPIO DECLARATION NODE
+ * ========================================================================= */
+/ {
+    user_gpios {
+        compatible = "gpio-keys";
+        pinctrl-names = "default";
+        pinctrl-0 = <&user_inputs_pins>;
+
+        /* --- 8 DIGITAL INPUTS --- */
+        in1 { label = "IN_1"; gpios = <&gpioa 8 GPIO_ACTIVE_LOW>; };  /* CN2 Pin 7  (PA8)  */
+        in2 { label = "IN_2"; gpios = <&gpiob 10 GPIO_ACTIVE_LOW>; }; /* CN2 Pin 8  (PB10) */
+        in3 { label = "IN_3"; gpios = <&gpiob 12 GPIO_ACTIVE_LOW>; }; /* CN2 Pin 10 (PB12) */
+        in4 { label = "IN_4"; gpios = <&gpiog 8 GPIO_ACTIVE_LOW>; };  /* CN2 Pin 11 (PG8)  */
+        in5 { label = "IN_5"; gpios = <&gpioi 5 GPIO_ACTIVE_LOW>; };  /* CN2 Pin 12 (PI5)  */
+        in6 { label = "IN_6"; gpios = <&gpiod 7 GPIO_ACTIVE_LOW>; };  /* CN2 Pin 13 (PD7)  */
+        in7 { label = "IN_7"; gpios = <&gpiog 15 GPIO_ACTIVE_LOW>; }; /* CN2 Pin 15 (PG15) */
+        in8 { label = "IN_8"; gpios = <&gpiof 1 GPIO_ACTIVE_LOW>; };  /* CN2 Pin 16 (PF1)  */
+
+        /* --- 8 DIGITAL OUTPUTS --- */
+        out1 { label = "OUT_1"; gpios = <&gpiof 0 GPIO_ACTIVE_HIGH>; };  /* CN2 Pin 18 (PF0)  */
+        out2 { label = "OUT_2"; gpios = <&gpiof 9 GPIO_ACTIVE_HIGH>; };  /* CN2 Pin 19 (PF9)  */
+        out3 { label = "OUT_3"; gpios = <&gpiof 8 GPIO_ACTIVE_HIGH>; };  /* CN2 Pin 21 (PF9) */
+        out4 { label = "OUT_4"; gpios = <&gpiof 4 GPIO_ACTIVE_HIGH>; };  /* CN2 Pin 22 (PF4)  */
+        out5 { label = "OUT_5"; gpios = <&gpiof 7 GPIO_ACTIVE_HIGH>; };  /* CN2 Pin 23 (PF7)  */
+        out6 { label = "OUT_6"; gpios = <&gpiof 6 GPIO_ACTIVE_HIGH>; };  /* CN2 Pin 24 (PF6) */
+        out7 { label = "OUT_7"; gpios = <&gpiof 3 GPIO_ACTIVE_HIGH>; };  /* CN2 Pin 26 (PF3) */
+        out8 { label = "OUT_8"; gpios = <&gpiof 15 GPIO_ACTIVE_HIGH>; }; /* CN2 Pin 27 (PF15) */
+        out9 { label = "OUT_9"; gpios = <&gpioa 13 GPIO_ACTIVE_HIGH>; }; /* CN2 Pin 26 (PA13) */
+        out10 { label = "OUT_10"; gpios = <&gpioh 7 GPIO_ACTIVE_HIGH>; };  /* CN2 Pin 27 (PH7) */
+    };
+};
+
